src/util.coffee

```diff
@@ -8,13 +8,15 @@ class VASTUtil
             else
                 # node mode, do not track (unit test only)
 
-    @resolveURLTemplates: (URLTemplates, variables) ->
+    @resolveURLTemplates: (URLTemplates, variables = {}) ->
         URLs = []
 
-        variables ?= {} # ["CACHEBUSTING", "random", "CONTENTPLAYHEAD", "ASSETURI", "ERRORCODE"]
-        unless "CACHEBUSTING" of variables
-            variables["CACHEBUSTING"] = Math.round(Math.random() * 1.0e+10)
-        variables["random"] = variables["CACHEBUSTING"] # synonym for Auditude macro
+        variables["ASSETURI"]     = @encodeURIComponentRFC3986(variables["ASSETURI"]) if variables["ASSETURI"]?
+        variables["CACHEBUSTING"] = Math.round(Math.random() * 1.0e+10) unless variables["CACHEBUSTING"]?
+        variables["TIMESTAMP"]    = (new Date).toISOString() unless variables["TIMESTAMP"]?
```

Comment:
As the `CACHEBUSTING`, `TIMESTAMP`, `random`, `RANDOM` and `ERRORCODE` variable might be defined outside of this function, and then might have an arbitrary value, it might be a good idea to `@encodeURIComponentRFC3986()` all of them. What do you think?